cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicatorBuilder.java

```diff
@@ -39,6 +39,46 @@
     private List<HttpConnectionResponseInterceptor> responseInterceptors = new ArrayList
             <HttpConnectionResponseInterceptor>();
 
+    private URI addCookieInterceptorIfRequired(URI uri) {
+        String urlProtocol = uri.getScheme();
+        String urlHost = uri.getHost();
+        String urlPath = uri.getRawPath();
+
+        //Check if port exists
+        int urlPort = uri.getPort();
+        if (urlPort < 0) {
+            if (urlProtocol.equals("http")) {
```

Comment:
This would be better written as `if("http".equals(urlProtocol)) {` so it can't NPE.
Similarly for the `else if` two lines later.